protocol/src/main/java/org/fido/iot/protocol/epid/EpidHttpClient.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Intel Corporation
+// SPDX-License-Identifier: Apache 2.0
+
+package org.fido.iot.protocol.epid;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class EpidHttpClient {
+
+  private static ExecutorService executor = Executors.newCachedThreadPool(r -> {
+    Thread t = Executors.defaultThreadFactory().newThread(r);
+    t.setDaemon(true);
+    return t;
+  });
+  private static final long httpRequestTimeout = Duration.ofSeconds(10).getSeconds();
+
+  /**
+   * Perform HTTP GET operation.
+   * 
+   * @param url the Url to which request will be sent
+   * @return the response from the Url
+   * @throws IOException throws an IOException in case the response status code is not 200 (OK)
+   */
+  public static byte[] doGet(String url) throws IOException {
+    HttpClient httpClient;
+    try {
+      httpClient = HttpClient.newBuilder().build();
+      final HttpRequest.Builder httpRequestBuilder =
+          HttpRequest.newBuilder().header("Accept", "application/octet-stream");
+      final HttpRequest httpRequest = httpRequestBuilder.uri(URI.create(url)).GET().build();
+      final Future<HttpResponse<byte[]>> future =
+          executor.submit(() -> httpClient.send(httpRequest, BodyHandlers.ofByteArray()));
+      final HttpResponse<byte[]> httpResponse;
+      httpResponse = future.get(httpRequestTimeout, TimeUnit.SECONDS);
+      if (httpResponse.statusCode() == 200) {
```

Comment:
What about other 200-family "OK" responses?

Also, my experience with this library is that non-2xx responses already generate IOExceptions.  Has the error path been tested?

Reply:
We've always checked for 200 only as far as i'm aware of.
Not sure if it generates an IOException by default for non-2xx series, since with a modified sample TO2 client, we see Timeout exception right now. This, specifically will be tested against valid device later.